mp4parse/src/lib.rs

```diff
@@ -4058,13 +4070,31 @@ fn read_meta<T: Read>(src: &mut BMFFBox<T>) -> Result<MetadataBox> {
     while let Some(mut b) = iter.next_box()? {
         match b.head.name {
             BoxType::MetadataItemListEntry => read_ilst(&mut b, &mut meta)?,
+            BoxType::MetadataXMLBox => read_xml_(&mut b, &mut meta)?,
+            BoxType::MetadataBXMLBox => read_bxml(&mut b, &mut meta)?,
             _ => skip_box_content(&mut b)?,
         };
         check_parser_state!(b.content);
     }
     Ok(meta)
 }
 
+/// Parse a XML box inside a meta box
+fn read_xml_<T: Read>(src: &mut BMFFBox<T>, meta: &mut MetadataBox) -> Result<()> {
+    let (_, _) = read_fullbox_extra(src)?;
```

Comment:
Since the spec requires both the flags and version fields to be 0, I'd like this to validate that:
```suggestion
    if read_fullbox_version_no_flags(src)? != 0 {
        return Err(Error::Unsupported("unsupported XmlBox version"));
    }
```
Similarly for `read_bxml`.